src/lib/inference-stack.ts

```diff
@@ -0,0 +1,95 @@
+import * as path from 'path';
+import { IVpc, ISecurityGroup, SecurityGroup } from '@aws-cdk/aws-ec2';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import { IFunction, Runtime } from '@aws-cdk/aws-lambda';
+import { PythonFunction, PythonLayerVersion } from '@aws-cdk/aws-lambda-python';
+import { IQueue } from '@aws-cdk/aws-sqs';
+import { Construct, Duration, Stack, NestedStack, NestedStackProps } from '@aws-cdk/core';
+
+export interface InferenceProps extends NestedStackProps {
+  readonly vpc: IVpc;
+  readonly neptune: {
+    endpoint: string;
+    port: string;
+    clusterResourceId: string;
+  };
+  readonly queue:IQueue;
+}
+
+export class InferenceStack extends NestedStack {
+  readonly inferenceSG: ISecurityGroup;
+  readonly inferenceStatsFn: IFunction;
+
+  constructor(scope: Construct, id: string, props: InferenceProps) {
+    super(scope, id, props);
+
+    const endpointName = 'FraudDetection'.toLowerCase();
+
+    this.inferenceStatsFn = new PythonFunction(this, 'InferenceStatsFn', {
+      entry: path.join(__dirname, '../lambda.d/inference/func'),
+      layers: [
+        new PythonLayerVersion(this, 'InferenceDataLayer', {
+          entry: path.join(__dirname, '../lambda.d/inference/layer'),
+          compatibleRuntimes: [Runtime.PYTHON_3_8],
+        }),
+        new PythonLayerVersion(this, 'InferenceNeptuneLibLayer', {
+          entry: path.join(__dirname, '../script-libs/amazon-neptune-tools/neptune-python-utils'),
+          compatibleRuntimes: [Runtime.PYTHON_3_8],
+        }),
+      ],
+      index: 'inferenceApi.py',
+      runtime: Runtime.PYTHON_3_8,
+      handler: 'handler',
+      timeout: Duration.minutes(2),
+      memorySize: 512,
+      environment: {
+        MAX_FEATURE_NODE: String(10),
+        CLUSTER_ENDPOINT: props.neptune.endpoint,
+        CLUSTER_PORT: props.neptune.port,
+        CLUSTER_REGION: String(this.region),
+        ENDPOINT_NAME: endpointName,
+        MODEL_BTW: String(0.5),
+        QUEUE_URL: props.queue.queueUrl,
+        TRANSACTION_ID_COLS: 'card1,card2,card3,card4,card5,card6,ProductCD,addr1,addr2,P_emaildomain,R_emaildomain',
```

Comment:
should share the column lists with training stack,

https://github.com/aws-samples/realtime-fraud-detection-with-gnn-on-dgl/blob/main/src/lib/etl-glue.ts#L190-L191